lib/matplotlib/axes/_base.py

```diff
@@ -275,10 +271,10 @@ def _setdefaults(self, defaults, *kwargs):
     def _makeline(self, x, y, kw, kwargs):
         kw = kw.copy()  # Don't modify the original kw.
         kwargs = kwargs.copy()
-        default_dict = self._getdefaults(None, kw, kwargs)
-        self._setdefaults(default_dict, kw, kwargs)
+        kw.update(kwargs)
```

Comment:
Most of those comments are on the `_makefill` related code which looks much worse.

In this case the user kwargs were just getting passed to `line.set` which just farms them back out to `line.set_*(...)`.  Now the same thing is happening (through `update` which does the same dispatching, but does some default cleaning first (which strips the `None`s)).